backend/codesirius/problems/views/problem.py

```
@@ -0,0 +1,132 @@
+            logger.info(f"Problem created successfully with ID: {problem.id}")
+            return CodesiriusAPIResponse(
+                data={"id": problem.id},
+                status_code=status.HTTP_201_CREATED,
+                message="Problem created",
+            )
+        logger.warning("Problem creation failed due to validation errors")
+        raise ValidationError(serializer.errors)
+
+
+class IsOwnerOrReadOnly(IsAuthenticatedOrReadOnly):
+    def has_object_permission(self, request, view, obj):
```

Ruff (0.8.2)

Missing return type annotation for public function `has_object_permission`

(ANN201)